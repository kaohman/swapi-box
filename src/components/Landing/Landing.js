import React from 'react';
import PropTypes from 'prop-types';

const Landing = ({ text, title, date }) => {

  return (
    <div id='landing-container'>
      <h3>{text}</h3>
      <h3>- {title}</h3>
      <h3>- {date}</h3>
      <h4>SELECT A CATEGORY</h4>
    </div>
  );
}

Landing.propTypes = {
  text: PropTypes.string,
  title: PropTypes.string,
  date: PropTypes.string
}

Landing.defaultProps = {
  text: 'DEFAULT TEXT',
  title: 'DEFAULT TITLE',
  date: '00-00-0000'
}


export default Landing;