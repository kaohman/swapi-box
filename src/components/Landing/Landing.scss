#landing-container {
  @include main-content-style();
}

// #scroll-text-container {
 
// }

.landing-text {
  text-align: justify;
  font-size: 40px;
  font-family: $content-font;
  color: $header-font-color;
}

.right-justify {
  text-align: right;
}


