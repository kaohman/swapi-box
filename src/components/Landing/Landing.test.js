import React from 'react';
import ReactDOM from 'react-dom';
import Landing from './Landing';
import { shallow } from 'enzyme';

const mockScrollText = {
  text: 'this is sample text',
  title: 'movie title',
  date: '11-5-2018'
}

describe('Landing', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <Landing
        {...mockScrollText}
      />
    );
  });

  it('should match the snapshot with all data passed in', () => {
    expect(wrapper).toMatchSnapshot();
  });
});