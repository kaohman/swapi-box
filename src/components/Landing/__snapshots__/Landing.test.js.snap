// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing should match the snapshot with all data passed in 1`] = `
<div
  id="landing-container"
>
  <div
    id="scroll-text-container"
  >
    <h3
      className="landing-text"
    >
      this is sample text
    </h3>
    <h3
      className="landing-text right-justify"
    >
      - 
      movie title
    </h3>
    <h3
      className="landing-text right-justify"
    >
      - 
      11-5-2018
    </h3>
  </div>
</div>
`;
